scripts/lib/file

```diff
@@ -46,24 +46,31 @@ spec:
 EOF
 }
 
-# create_operator_group_file creates an operator group for the operator CSV.
+# create_operator_group_file creates an operator group for the operator CSV if
+# the CSV does not support AllNamespaces.
 function create_operator_group_file() {
   local op_group_file="$1"
-  local name="$2"
-  local service_account="$3"
+  local op_name="$2"
+  local csv_file="$3"
   local namespace="$4"
 
-  cat <<EOF > "$op_group_file"
+  while read -r mode; do
+    if [[ "$(echo "$mode" | jq -r ".type")" == "AllNamespaces" && \
+      "$(echo "$mode" | jq -r ".supported")" == "false" ]]; then
+        local service_account="$(yq r "$csv_file" "spec.install.spec.deployments[0].spec.template.spec.serviceAccountName")"
+        cat <<EOF > "$op_group_file"
 apiVersion: operators.coreos.com/v1alpha2
 kind: OperatorGroup
 metadata:
-  name: ${name}-og
+  name: ${op_name}-og
   namespace: $namespace
 spec:
   serviceaccount: $service_account
   targetNamespaces:
   - $namespace
 EOF
+    fi
+  done < <(yq r --tojson "$csv_file" "spec.installModes" | jq -c '.[]')
```

Comment:
I'm not sure what this is doing

Reply:
Grabbing `installModes` from the CSV and iterating through them.

Reply:
Is this used in this function? Is an operator group being created for each installMode?

Reply:
No, check out [line 58](https://github.com/operator-framework/community-operators/pull/205/files/179920be0ff1c1f2a8d1e65dccdbc9a0b41520ad#diff-a7cef9f7c433865159effd45b37f4a6aR58). We're parsing `installModes` to check whether `AllNamespaces` is not supported, and if not, create the `OperatorGroup`. Otherwise do not create one.

Reply:
Awesome, totally missed that. I wish it was a bit more straightforward, but that's just a nitpick